core/client/fs/src/main/java/alluxio/client/file/cache/TimeLimitedPageStore.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ *
+ */
+
+package alluxio.client.file.cache;
+
+import alluxio.exception.PageNotFoundException;
+import alluxio.metrics.MetricKey;
+import alluxio.metrics.MetricsSystem;
+
+import com.codahale.metrics.Counter;
+import com.google.common.base.Throwables;
+import com.google.common.util.concurrent.SimpleTimeLimiter;
+import com.google.common.util.concurrent.TimeLimiter;
+
+import java.io.IOException;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Stream;
+
+/**
+ * A wrapper class on PageStore with timeout.
+ */
+public class TimeLimitedPageStore implements PageStore {
+  private final PageStore mPageStore;
+  private final long mTimeoutMs;
+  private final TimeLimiter mTimeLimter;
+
+  /**
+   * @param pageStore page store
+   * @param timeout time out in ms
+   */
+  public TimeLimitedPageStore(PageStore pageStore, long timeout) {
+    mPageStore = pageStore;
+    mTimeoutMs = timeout;
+    mTimeLimter = SimpleTimeLimiter.create(Executors.newCachedThreadPool());
+  }
+
+  @Override
+  public void put(PageId pageId, byte[] page) throws IOException {
+    try {
+      Callable<Void> callable = () -> {
+        mPageStore.put(pageId, page);
+        return null;
+      };
+      mTimeLimter.callWithTimeout(callable, mTimeoutMs, TimeUnit.MILLISECONDS);
```

Comment:
Do the corresponding page stores need to handle `InterruptedException`? For example, `LocalPageStore` will throw an `IOException` if it catches any exception, so we will not see `TimeoutException`?

Reply:
we will need to handle `InterruptedException`  her. Updating the code